src/edu/csus/ecs/pc2/clics/API202306/StateService.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 1989-2024 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import javax.inject.Singleton;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Feature;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.ext.Provider;
+
+import edu.csus.ecs.pc2.core.IInternalController;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * WebService to handle "state" REST endpoint as described by the CLICS wiki.
+ *
+ * @author pc2@ecs.csus.edu
+ *
+ */
+@Path("/contests/{contestId}/state")
+@Produces(MediaType.APPLICATION_JSON)
+@Provider
+@Singleton
+public class StateService implements Feature {
+
+    private IInternalContest model;
+
+    private IInternalController controller;
+
+    public StateService(IInternalContest inModel, IInternalController inController) {
+        super();
+        this.model = inModel;
+        this.controller = inController;
+    }
+
+    /**
+     * This method returns a representation of the current contest state in JSON format as described on the CLICS wiki.
+     *
+     * @param sc user information
+     * @param contestId The contest
+     * @return a {@link Response} object containing a JSON String giving the scheduled contest start time as a Unix Epoch value, or as the string "undefined" if no start time is currently scheduled.
```

Comment:
I don't think this is an accurate description of what's "returned".  The returned JSON doesn't contain "scheduled contest start time"; it contains "contest state" including _actual_ start time (or null), plus a lot of other stuff.